org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -263,6 +264,42 @@ public long getStartVersion() {
 			return this.startVersion;
 		}
 
+		public SingleServerDocumentExecutor toExecutor(final @NonNull LanguageServerWrapper serverWrapper) {
+			return new SingleServerDocumentExecutor(this.document, serverWrapper);
+		}
+	}
+
+	/**
+	 * Executor for a single server - for follow-up requests that should go to the server that returned an earlier response
+	 *
+	 */
+	public static class SingleServerDocumentExecutor extends LanguageServerDocumentExecutor {
+
+		private final @NonNull LanguageServerWrapper serverWrapper;
+
+		SingleServerDocumentExecutor(final @NonNull IDocument document, final @NonNull LanguageServerWrapper serverWrapper) {
+			super(document);
+			this.serverWrapper = serverWrapper;
+		}
+
+		@Override
+		@SuppressWarnings("null")
+		protected @NonNull List<@NonNull CompletableFuture<@Nullable LanguageServerWrapper>> getServers() {
+			// Compute list of servers from document & filter
+			return Collections.singletonList(CompletableFuture.completedFuture(this.serverWrapper));
```

Comment:
just style, but I think that `serverWrapper` reads better than `this.serverWrapper`

Reply:
Sorry, force of habit!